src/utils.ts

```diff
@@ -1,6 +1,38 @@
-export function divideAndRound(dividend: bigint, divisor: bigint) {
-  const invertSign = dividend < 0n ? -1n : 1n;
-  return (dividend * invertSign + divisor / 2n) / divisor * invertSign;
+import { Rounding } from "./types";
+
+function divideAndRoundUp(dividend: bigint, divisor: bigint) {
+  const num = divisor > 0n ? dividend : -dividend;
+  const den = divisor > 0n ? divisor : -divisor;
+  const remainder = num % den;
+  const roundUp = remainder > 0n ? 1n : 0n;
+  return num / den + roundUp;
+}
+
+function divideAndRoundDown(dividend: bigint, divisor: bigint) {
+  const num = divisor > 0n ? dividend : -dividend;
+  const den = divisor > 0n ? divisor : -divisor;
+  const remainder = num % den;
+  const roundDown = remainder < 0n ? -1n : 0n;
+  return num / den + roundDown;
+}
+
+function divideAndRoundHalf(dividend: bigint, divisor: bigint) {
+  const num = divisor > 0n ? dividend : -dividend;
+  const den = divisor > 0n ? divisor : -divisor;
+  const invertSign = num < 0n ? -1n : 1n;
+  return (num * invertSign + den / 2n) / den * invertSign;
+}
```

Comment:
Should we pass `num` and `den` directly to these three functions (for bundle size)? It shouldn’t be much of an issue since we don’t expose these functions.

Reply:
We could because it's common code but the reason I didn't is because the call to these three functions would return unexpected results if the denominator were negative. Let me know what you think

Reply:
Cool let’s keep it this way then, thanks for the explanation :+1: